### ISSUE TITLE
Exception if running in console

### ISSUE THREAD

**Issue by iaK** [livewire/livewire#298]
I've got a static site generator and trying to use Livewire in it. 
However, it crashes when i try to build the views.

It throws:

```
Facade\Ignition\Exceptions\ViewException  : Call to a member function gatherMiddleware() on null

in

vendor/livewire/livewire/src/LivewireManager.php:123
```

It's trying to get the middleware fromt the request, which does not exist in the console context.

```php
if (app()->runningUnitTests()) {
    // There is no "request->route()" to access in unit tests.
    return [];
}

return request()->route()->gatherMiddleware();
```

I don't really understand what this is used for, is it possible to build the views without a request, or can't I use Livewire in this context?

**Comment by iaK** [comment]
There is a method that might work - app()->runningInConsole() - to avoid the error. But the question is what livewire does with the middleware stack. i can see it is encoded and sent to the frontend, but there i loose track of what really happens with it.

**Comment by iaK** [comment]
Okay, done some more digging. It seems the middlewares are applied manually by Livewire.

Without knowing much about the reason behind thid- Could you maybe (optionally ofc) manually enter which middlewares should be applied to the request via the config file?

**Comment by adrianschubek** [comment]
It's already possible to disable middleware in Livewire completely. Edit your `AppServiceProvider` like so
```php
public function boot()
{
    Livewire::filterMiddleware(function($middleware) {
        return false;
    });
}
```

**Comment by iaK** [comment]
Thanks, didn't know that :) it still throws the error thought.

**Comment by calebporzio** [comment]
This could pretty easily be made possible. Anyone open to PRing a fix for it? You can just skip the middware thing if running in console.

**Comment by lancepioch** [comment]
This looks to have been resolved by 5ea11e7d4c3ab0489b5004e9c2c77b89c5c7a9c6

If you're still having issues, ping me with your code/lib and I'll look through the issue and send a pr to resolve it.
